pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,127 @@ import (
 }
 
-// Converter returns the api.Scheme for the API version to test against, as set by the
-// KUBE_API_VERSION env var.
-func Converter() runtime.ObjectConvertor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// Converter returns the api.Scheme for the API version of the group to test
+// against, as set by the KUBE_TEST_API env var.
+func Converter(group ...string) runtime.ObjectConvertor {
+	if len(group) == 0 || group[0] == "" {
+		interfaces, err := latest.InterfacesFor(Version())
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
+	}
+	if group[0] == "experimental" {
+		interfaces, err := explatest.InterfacesFor(Version(group...))
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
 	}
-	return interfaces.ObjectConvertor
+	panic(fmt.Errorf("cannot test group %s", group))
 }
 
-// MetadataAccessor returns the MetadataAccessor for the API version to test against,
-// as set by the KUBE_API_VERSION env var.
-func MetadataAccessor() meta.MetadataAccessor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// MetadataAccessor returns the MetadataAccessor for the API version of the
+// group to test against, as set by the KUBE_TEST_API env var.
+func MetadataAccessor(group ...string) meta.MetadataAccessor {
+	if len(group) == 0 || group[0] == "" {
+		interfaces, err := latest.InterfacesFor(Version())
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
+	}
+	if group[0] == "experimental" {
+		interfaces, err := explatest.InterfacesFor(Version(group...))
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
 	}
-	return interfaces.MetadataAccessor
+	panic(fmt.Errorf("cannot test group %s", group))
 }
 
-// SelfLink returns a self link that will appear to be for the version Version().
+// SelfLink returns a self link that will appear to be for the version
+// Version(group...).
 // 'resource' should be the resource path, e.g. "pods" for the Pod type. 'name' should be
 // empty for lists.
-func SelfLink(resource, name string) string {
+func SelfLink(resource, name string, group ...string) string {
 	if name == "" {
-		return fmt.Sprintf("/api/%s/%s", Version(), resource)
+		return fmt.Sprintf("/api/%s/%s", Version(group...), resource)
 	}
-	return fmt.Sprintf("/api/%s/%s/%s", Version(), resource, name)
+	return fmt.Sprintf("/api/%s/%s/%s", Version(group...), resource, name)
 }
 
 // Returns the appropriate path for the given prefix (watch, proxy, redirect, etc), resource, namespace and name.
 // For ex, this is of the form:
 // /api/v1/watch/namespaces/foo/pods/pod0 for v1.
-func ResourcePathWithPrefix(prefix, resource, namespace, name string) string {
-	path := "/api/" + Version()
+// TODO: needs to change the prefix for API groups other than v1.
+func ResourcePathWithPrefix(prefix, resource, namespace, name string, group ...string) string {
```

Comment:
Please make the order `prefix, group, resource, namespace, name`. This one definitely don't let group be optional; it shouldn't be used too many places?